src/dbnode/storage/index/block.go

```diff
@@ -424,82 +424,99 @@ func (b *block) Query(
 	results DocumentResults,
 	logFields []opentracinglog.Field,
 ) (bool, error) {
-	ctx, sp := ctx.StartTraceSpan(tracepoint.BlockQuery)
-	sp.LogFields(logFields...)
-	defer sp.Finish()
-
 	start := time.Now()
-	exhaustive, err := b.queryWithSpan(ctx, query, opts, results)
+	iter, err := b.QueryIter(ctx, query)
 	if err != nil {
-		sp.LogFields(opentracinglog.Error(err))
+		return false, err
+	}
+	iterCloser := safeCloser{closable: iter}
+	defer func() {
+		_ = iterCloser.Close()
+		b.metrics.queryDocsMatched.RecordValue(float64(results.TotalDocsCount()))
+		b.metrics.querySeriesMatched.RecordValue(float64(results.Size()))
+	}()
+	if err := b.QueryWithIter(ctx, opts, iter, results, logFields); err != nil {
```

Comment:
in the next PR, `block.Query` will removed and the index worker will have this code.